docs/en/src/writing-plugins.md

```diff
@@ -51,6 +50,18 @@ to append `.xplr` to the name to make them distinguishable. Similar to the
 Finally, after publishing, don't hesitate to
 [let us know][4].
 
+## Best practices
+
+- Try not to execute a lot of commands at startup, it may make xplr slow to
+  start.
+- When executing commands, prefer `Call0` over `Call`, `BashExec0` over
+  `BashExec` and so on. File names may contain newline characters
+  (e.g. `foo$'\n'bar`).
+- File names may also contain quotes. Use the syntax
+  `printf 'Call0: "%s"' "${FOO_ESC}"` where `FOO_ESC=${FOO//\"/\\\"}`.
```

Comment:
Depending on how this is used, it may be easier to write:
```sh
printf 'Call0: %q' "$file"
```

The `%q` placeholder will shell escape input, so if:
```sh
file="foo \"bar\"
baz 'qux'"
```
Using `%q` would output a shell safe string like `$'foo "bar"\nbaz \'qux\''`